types/store.go

```diff
@@ -263,7 +277,24 @@ func PrefixEndBytes(prefix []byte) []byte {
 	return end
 }
 
+// Getter struct for prefixed stores
+type PrefixStoreGetter struct {
```

Comment:
Can you explain why we need this, as opposed to calling `ctx.KVStore(storeKey).Prefix(prefix)`, which doesn't seem to be any more onerous?

Reply:
In that case, we are passing the storeKey to external libraries. There is no guarantee that the libraries will use it as prefixed form. Passing PrefixStoreGetters to external libraries is a safe way to allow the libraries to access only the prefixed space of that storeKey.

Reply:
Do you mean external modules? Why do they need to access the same store?

(or if not, what external libraries are you referencing?)

Reply:
For example, Oracle module takes KVStoreGetter(mostly will be PrefixStoreGetter) and stores Oracle related information there. Modules those use Oracle only have to pass sdk.NewPrefixStoreGetter(storeKey), which will wrap the storeKey and generate a safe accessor to the limited prefixed space. So even if there is an unexpected behavior in Oracle, it cannot modify the outside of its permitted space.

Reply:
OK, thanks for the explanation!